homeassistant/components/todoist/calendar.py

```diff
@@ -426,26 +477,39 @@ def calendar_event(self) -> CalendarEvent | None:
                 start=start,
                 end=start + timedelta(days=1),
             )
-        return CalendarEvent(
-            summary=self.event[SUMMARY], start=self.event[START], end=self.event[END]
-        )
+        if end := self.event[END]:
```

Comment:
This new guard introduced is a bit hard to follow. Consider structuring another way so this `return None` is not needed given that I believe that is a shouldn't happen case.  It makes it appear like it's possible but its not unless i missed something (and its changing behavior)

Reply:
I believe this was a mypy error I was addressing, but it's been a few months since I made these changes.  

`CalanderEvent` requires a date or datetime for the `end` parameter.  The block above has 2 checks in the or

`if not self.event.get(END) or self.event.get(ALL_DAY):`

So by the time we get to the new type gaurd check if the right hand side of the `or` was True, that could mean either it's an all day event OR the event END key was `None`.  Therefore mypy wanted an extra check so that we did't try to pass a None value for `end` to the `CalendarEvent` object.  

Hopefully breaking out the 2 checks simplifies things more.

Reply:
I had the impression generally that mypy is smart enough to detect that kind of issue.